### ISSUE TITLE
Support grafana-like overrides

### ISSUE THREAD

**Issue by 7840vz** [perses/perses#2169]
### Proposal

This PR introduces color overrides by query https://github.com/perses/perses/pull/2046

In our grafana dashboards, we rely heavily on panel [overrides](https://grafana.com/docs/grafana/latest/panels-visualizations/configure-overrides/), that can override not only colors, but also any visualization options (including line transparency, stacking, units etc). Is there any way to support overrides in Perses in a way, that any visualization option can be overriden? This could help to migrate our dashboards from Grafana.

**Comment by Nexucis** [comment]
@AntoineThebaud @Gladorme @celian-garcia @jgbernalp any opinions here ?

**Comment by abelyakin** [comment]
I completely agree.  Adding more configuration possibilities on per query basis to mimic Grafana’s “overrides” functionality is a great idea. I’d be happy to help with the implementation. I’d like to suggest to start extending the functionality further by adding  properties from “visual” block, similar to how color is currently handled. 

Let me know what you think, and I’d be glad to contribute to this effort.

**Comment by AntoineThebaud** [comment]
Yes there's no reason we should stop at color overriding and not propose further customization per query. #2046 actually prepared the ground for this, building on top should be quite easy.
Still, in terms of customization in general we were several to think that we should progress carefully, because we feel grafana became quite messy over the years after adding more and more customization capabilities.
About this question precisely _"Is there any way to support overrides in Perses in a way, that any visualization option can be overriden?"_, I think the options should be looked at case by case, e.g I see no issue adding support for transparency and/or line sickness overriding, but regarding e.g stacking & units overrides I have the impression the resulting graph would be hardly understandable. @7840vz do you have any example to provide here?

**Comment by 7840vz** [comment]
Actually, from my experience with Grafana, I can think of dozens of examples where you need stacking & units overrides.

Units: if two Y axis are supported, you can map different series to different axis and use one units for one and second one for another one. 
Stacking 1: one of series could return threshold value (like max memory available), while another series could return memory types used. You could think that stacking memory used is a good idea, while keeping threshold away.

Also, in Grafana we use DAC approach a lot, and thanks to overrides we could keep actual Panel visualisation clean with defaults and scope(glue) any visuals stuff together with metric(query) itself:
- what units to use for this metric
- how we want to draw it (like lines or bars etc)
So in case we add two metrics(queries) to the single panel there is no overlap and no conflicts. 

Another good example for overrides is table chart, where each query could be a column. again, if units are scoped into override of the specific query, it won't be shown inside columns of other charts.


While you could think of some overrides as not useful, they actually could be in some cases, so I would rather allow anything to be overriden for flexibility and , just like in Grafana.

**Comment by AntoineThebaud** [comment]
Btw https://github.com/perses/plugins/pull/386 implemented opacity & style overrides at query-level. I will open dedicated issues for the 2 other customizations discussed here (stacking & unit). Please open further issues for any other option you would like to override at query level
